packages/dev/core/src/Materials/Node/nodeMaterial.ts

```diff
@@ -855,6 +855,28 @@ export class NodeMaterial extends PushMaterial {
             this._initializeBlock(fragmentOutputNode, this._fragmentCompilationState, vertexNodes, autoConfigure);
         }
 
+        // Are blocks code ready?
+        let waitingNodeCount = 0;
+        for (const node of this.attachedBlocks) {
+            if (!node.codeIsReady) {
+                waitingNodeCount++;
+                node.onCodeIsReadyObservable.addOnce(() => {
+                    waitingNodeCount--;
+                    if (waitingNodeCount === 0) {
+                        this._finishBuildProcess(verbose, updateBuildId, vertexNodes, fragmentNodes);
+                    }
+                });
+            }
+        }
+
+        if (waitingNodeCount !== 0) {
+            return;
+        }
+
+        this._finishBuildProcess(verbose, updateBuildId, vertexNodes, fragmentNodes);

```

Comment:
I think there's a problem here, because before the change with the dynamic imports, when `build` returned, the material was really built, whereas now it is not / may be not anymore. I guess it could be considered a breaking change, depending on what a user does after `build` has been called.

Maybe we could add a `buildAsync` that would resolve only when everything is really built, and document this change.